cns/service/main.go

```diff
@@ -506,9 +530,7 @@ func main() {
 	// Initialze state in if CNS is running in CRD mode
 	// State must be initialized before we start HTTPRestService
 	if config.ChannelMode == cns.CRD {
-		requestControllerStopChannel := make(chan struct{})
-		defer close(requestControllerStopChannel)
-		err = InitializeCRDState(httpRestService, cnsconfig, requestControllerStopChannel)
+		err = InitializeCRDState(ctx, httpRestService, cnsconfig)
```

Comment:
These are great fixes but making this PR really bulky. Is it possible to move this to a new PR? If not, then may be in future we should take care of it.